### ISSUE TITLE
The training loss is big and can't be reduced

### ISSUE THREAD

**Issue by 1397201241** [lukasHoel/3DGS-LM#13]
Hello,

I encountered an issue while training on my custom dataset containing 79 images using a 32GB GPU. because OOM, so I set the -r 4 flag and the training process started without crashing, but the training loss is big a lot and did not decrease over time.

This behavior is unusual since the same setup works normally with 3D Gaussian Splatting (3DGS).
Below are the CLI command and logs for reference:
**3dgs**
<img width="1898" height="189" alt="Image" src="https://github.com/user-attachments/assets/7dc45e99-0b79-4880-ae0d-62cfdc8fbb4f" />
**3dgs-LM**

<img width="576" height="66" alt="Image" src="https://github.com/user-attachments/assets/594def62-f7c9-41af-8dfa-8e99b31a129a" />

<img width="596" height="67" alt="Image" src="https://github.com/user-attachments/assets/d414b754-dfc8-43a9-bef7-56feecae7fdc" />

Could you please help me understand why the model isn't converging? Thanks in advance!

**Comment by lukasHoel** [comment]
Hi, how many image batches per LM step do you use and how many images are within each batch? How many initialization iterations do you perform? Please note that the loss that we log here is not a mean but the sum of squared residuals to follow the definition of Gauss-Newton energy functions. In other words the loss appears to be a high number but the number still goes down.

**Comment by 1397201241** [comment]
Thanks for replies! As mentioned above, I modified parameter "-r" to 4 and used the default configs for all other parameters.

**Comment by lukasHoel** [comment]
Ah I see. Yeah sorry about the confusion, but the default parameters probably are not a good choice (we should update them).

Can you try the ones from the readme?

```
python train.py \
-s /path/to/custom_data \
--root_out outputs/custom_data \
--exp_name "custom_data" \
--eval \
--images "images" \
--resolution 4 \
--num_sgd_iterations_before_gn 20000 \
--perc_images_in_line_search 0.3 \
--pcg_rtol 5e-2 \
--pcg_max_iter 8 \
--iterations 5 \
--test_iterations 5 \
--save_iterations 5
```

Note that different from the readme, I left out these arguments, using the default parameters. It's better to use no image subsampling and one iteration in your case, as the whole dataset fits in 15GB memory according to the logs.

```
--image_subsample_size -1 \
--image_subsample_n_iters 1 \
--image_subsample_frame_selection_mode "strided" \
```

What is the final psnr that you reach?

**Comment by 1397201241** [comment]
Okay, I will rerun the program according to your suggestions. Additionally, since the logs indicated that the task would take about five days to complete and GPU resources are currently limited, I decided to terminate the job. As a result, I wasn’t able to obtain the final PSNR value—sorry about that.

**Comment by lukasHoel** [comment]
You should not run 30k lm steps. Try the suggested values from the paper: 20k init iters and then 5-10 LM, specified with --iterations 5

**Comment by 1397201241** [comment]
Following your suggestions, I re-ran the program, but it ran into an Out Of Memory (OOM) error. As a result, I terminated the execution and adjusted the command to: 
`
python train.py -s /data/sfms/garden/ --exp_name "garden" --eval --images "images_8" -r 8 --num_sgd_iterations_before_gn 20000 --perc_images_in_line_search 0.3 --pcg_rtol 5e-2 --pcg_max_iter 8 --iterations 5 --test_iterations 5 --save_iterations 5 --image_subsample_size -1 --image_subsample_n_iters 1 --image_subsample_frame_selection_mode "strided"
`
The run completed successfully after the change, but the exported model performs very poorly. I'm not sure what went wrong.

**Comment by 1397201241** [comment]
> Following your suggestions, I re-ran the program, but it ran into an Out Of Memory (OOM) error. As a result, I terminated the execution and adjusted the command to: `python train.py -s /data/sfms/garden/ --exp_name "garden" --eval --images "images_8" -r 8 --num_sgd_iterations_before_gn 20000 --perc_images_in_line_search 0.3 --pcg_rtol 5e-2 --pcg_max_iter 8 --iterations 5 --test_iterations 5 --save_iterations 5 --image_subsample_size -1 --image_subsample_n_iters 1 --image_subsample_frame_selection_mode "strided"` The run completed successfully after the change, but the exported model performs very poorly. I'm not sure what went wrong.

**Comment by 1397201241** [comment]
> Following your suggestions, I re-ran the program, but it ran into an Out Of Memory (OOM) error. As a result, I terminated the execution and adjusted the command to: `python train.py -s /data/sfms/garden/ --exp_name "garden" --eval --images "images_8" -r 8 --num_sgd_iterations_before_gn 20000 --perc_images_in_line_search 0.3 --pcg_rtol 5e-2 --pcg_max_iter 8 --iterations 5 --test_iterations 5 --save_iterations 5 --image_subsample_size -1 --image_subsample_n_iters 1 --image_subsample_frame_selection_mode "strided"` The run completed successfully after the change, but the exported model performs very poorly. I'm not sure what went wrong.

 Failed to upload "image.png" 
 ：（
